server/sonar-server/src/main/java/org/sonar/server/permission/PermissionPrivilegeChecker.java

```diff
@@ -42,9 +43,28 @@ public static void checkProjectAdminUserByComponentKey(UserSession userSession,
     }
   }
 
-  public static void checkProjectAdminUserByComponentUuid(UserSession userSession, @Nullable String componentUuid) {
+  /**
+   * Checks that user is administrator of the specified project, or of the specified organization if project is not
+   * defined.
+   * @throws org.sonar.server.exceptions.ForbiddenException if user is not administrator
+   */
+  public static void checkProjectAdmin(UserSession userSession, String organizationUuid, Optional<ProjectId> projectId) {
     userSession.checkLoggedIn();
-    if (componentUuid == null || !userSession.hasComponentUuidPermission(UserRole.ADMIN, componentUuid)) {
+    if (!projectId.isPresent() || !userSession.hasComponentUuidPermission(UserRole.ADMIN, projectId.get().getUuid())) {
+      userSession.checkOrganizationPermission(organizationUuid, GlobalPermissions.SYSTEM_ADMIN);
+    }
+  }
+
+  /**
+   * Checks that user is administrator of the specified project, or of system if project is not
+   * defined.
+   * @throws org.sonar.server.exceptions.ForbiddenException if user is not administrator
+   * @deprecated does not support organizations. Replaced by {@link #checkProjectAdmin(UserSession, String, Optional)}
+   */
+  @Deprecated
+  public static void checkProjectAdmin(UserSession userSession, Optional<ProjectId> projectId) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Specify a "ProjectId" parameter instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS3553)
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)